catacomb/catacomb.go

```diff
@@ -112,10 +114,12 @@ func Invoke(plan Plan) (err error) {
 	// This goroutine runs the work func and stops the catacomb with its error;
 	// and waits for for the listen goroutine and all added workers to complete
 	// before marking the catacomb's tomb Dead.
-	catacomb.tomb.Go(func() error {
-		defer catacomb.wg.Wait()
-		catacomb.Kill(runSafely(plan.Work))
-		return nil
+	pprof.Do(catacomb.scopedContext(), pprof.Labels("catacomb-work", funcName(plan.Work)), func(ctx context.Context) {
```

Comment:
Invoke already takes a Plan object, which I think we could extend with a human-readable override. We could still use "funcName" when plan doesn't have a human label, but would it make things clearer for us conceptually to be able to supply the annotation?

I don't want to get too complex (for example, we could provide a way to do a generic number of pprof.Labels that get applied))
It may also be better to leave it as reflect, because then *everything* follows the same pattern, and humans will need to learn that pattern, but it avoids learning *two* patterns :)